mlflow/models/model.py

```diff
@@ -1172,7 +1174,9 @@ def log_model_metrics_for_step(client, model_id, run_id, step):
                     tags={key: str(value) for key, value in tags.items()}
                     if tags is not None
                     else None,
+                    flavor=flavor.__name__ if hasattr(flavor, "__name__") else "custom",
                 )
+                _last_logged_model_id.set(model.model_id)
```

Comment:
```suggestion
                record_telemetry(flavor=flavor)
                _last_logged_model_id.set(model.model_id)
```

I'm not sure if this works, but if the goal is collect data for what flavor the logged model is, do we need to decorate `create_logged_model`?

Reply:
This should work, but this means we need to add this line of code in several places (e.g. initialized_logged_model, create_external_model, client.create_logged_model...)

Reply:
I thought we track `create_logged_model` to collect flavor data, but we're also curious whether `create_logged_model` is used?